.github/workflows/_scheduled-audit.yml

```diff
@@ -57,13 +57,44 @@ jobs:
           set +e
           make audit
           EXIT_CODE=$?
-          # Provide heartbeat to bettersqtack for monitoring/alerting if heartbeat url is configured as secret
+          # Provide heartbeat to BetterStack for monitoring/alerting if heartbeat url is configured as secret
           if [ -n "$BETTERSTACK_AUDIT_HEARTBEAT_URL" ]; then
-            if [ $EXIT_CODE -eq 0 ]; then
-              curl -s $BETTERSTACK_AUDIT_HEARTBEAT_URL
-            else
-              curl -s $BETTERSTACK_AUDIT_HEARTBEAT_URL/$EXIT_CODE
-            fi
+            BETTERSTACK_METADATA_PAYLOAD=$(jq -n \
+              --arg github_workflow "${{ github.workflow }}" \
+              --arg github_run_url "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" \
+              --arg github_run_id "${{ github.run_id }}" \
+              --arg github_job "${{ github.job }}" \
+              --arg github_sha "${{ github.sha }}" \
+              --arg github_actor "${{ github.actor }}" \
+              --arg github_repository "${{ github.repository }}" \
+              --arg github_ref "${{ github.ref }}" \
+              --arg job_status "${{ job.status }}" \
+              --arg github_event_name "${{ github.event_name }}" \
+              --arg timestamp "$(date -u +"%Y-%m-%dT%H:%M:%SZ")" \
+              '{
+                github: {
+                  workflow: $github_workflow,
+                  run_url: $github_run_url,
+                  run_id: $github_run_id,
+                  job: $github_job,
+                  sha: $github_sha,
+                  actor: $github_actor,
+                  repository: $github_repository,
+                  ref: $github_ref,
+                  event_name: $github_event_name
+                },
+                job: {
+                  status: $job_status,
+                },
+                timestamp: $timestamp,
+              }'
+            )
+
+            curl "${BETTERSTACK_AUDIT_HEARTBEAT_URL}/${EXIT_CODE}" \
+              --silent \
+              --request POST \
+              --header "Content-Type: application/json" \
+              --data-binary "${BETTERSTACK_METADATA_PAYLOAD}"
```

Comment:
Probably `--fail-with-body` should be used? This will make the pipeline fail if there's a >=400 status code

Reply:
Yeah not a bad idea to add it in general, don't see any reason to not do so :tada: